src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -252,6 +256,32 @@ public sealed partial class TokenCache : ITokenCacheInternal
             }
         }
 
+        //This method pulls all of the acceess and refresh tokens from the cache and can therefore be very impactful on performance.
+        private void DumpCacheToLogs(AuthenticationRequestParameters requestParameters)
+        {
+            if (requestParameters.RequestContext.Logger.IsLoggingEnabled(LogLevel.Verbose))
+            {
+                IReadOnlyList<MsalAccessTokenCacheItem> accessTokenCacheItems = GetAllAccessTokensWithNoLocks(filterByClientId: false);
+                IReadOnlyList<MsalRefreshTokenCacheItem> refreshTokenCacheItems = GetAllRefreshTokensWithNoLocks(filterByClientId: false);
+                IList<MsalAccessTokenCacheKey> accessTokenCacheKeys = accessTokenCacheItems.Select(item => item.GetKey()).ToList();
+
+                requestParameters.RequestContext.Logger.Verbose($"Total number of access tokens in cache: {accessTokenCacheItems.Count()}");
```

Comment:
You can make it part of StringBuilder to log in one call. IMO, it will save time.